lib/pacemaker/pcmk_sched_colocation.c

```diff
@@ -1484,6 +1484,25 @@ pcmk__apply_coloc_to_priority(pcmk_resource_t *dependent,
                     pcmk_readable_score(dependent->priv->priority),
                     ((score_multiplier == 1)? "adding" : "subtracting"),
                     colocation->score);
+
+    if (pcmk__is_group(dependent->priv->parent)) {
+        /* For cloned group instances, we called this function for the first
+         * group member. Update the priority of the group itself for the purpose
```

Comment:
only true if `pcmk__group_colocated` is set (which will always be true by 3.0.0, but this will get backported to 2.1)

This belongs in the group method

I would consider this a fix (vs Low) and should have a regression test

Reply:
There's a lot of logic that would need to be duplicated in the group method since we don't have access to the score multiplier and whether we returned early or applied the colocation score... maybe I can just save the first member's priority beforehand and take the diff...

Reply:
Moving it to the group method seems messy in edge cases. Without a bunch of duplication, we can't tell whether we applied the colocation or not, all we can do is get the delta of the first member's priority.

If the first member starts out with priority `-900000` and the group starts out with priority `0` (due to meta-attributes or whatever), and we apply a colocation with score `-200000`, then the first member would end up at `-INFINITY` with a diff off `-100000`. Subtracting `100000` from the group's priority would be wrong: we want to subtract `200000`.

Correct me if you think I'm wrong here. If not, let me know whether you'd rather leave the fix in `pcmk__apply_coloc_to_priority()` or do however much refactoring or duplication is necessary to put it in the group method accurately.

Reply:
The methods are internal, so we could make it return the delta. If it would take a lot more than that, don't worry about it.

Reply:
That might work. The caller will have no way of knowing whether a colocation with score=0 was applied, but that would only be for tracing as of right now and wouldn't affect behavior, so not that important.

Reply:
Done in the "[(Likely squash into previous commit)](https://github.com/ClusterLabs/pacemaker/pull/3536/commits/44a55b9563be09f5e4b4fa1348baa425d4962c96)" commit. Leaving both for the moment so that you can compare them.